### PR TITLE
Refactor and test structured data

### PR DESCRIPTION
**Pull request by barbaraperic**
## Note
- I've added structured data for a [BreadcrumbList](https://developers.google.com/search/docs/appearance/structured-data/breadcrumb), which is commonly used in combination with WebPage (as the test result was not passing without it).
- [Ticket link](https://www.notion.so/filecoin/FF-V4-Refactor-Test-Structured-Data-ef5c22dc74874737921e6d419abb075c?pvs=4)

### Home Page
Home

### About Page
About

### Blog Page
Blog

### Blog Post Page
Blog Post

### Ecosystem Page
Ecosystem

### Ecosystem Post Page
Ecosystem Post

### Event Post Page
Event Post

### Event Page
Events

### Governance Page
Governance

### Grants Page
Grants

### Terms Page
Terms

### Privacy Policy Page
Privacy Policy

### CODE REVIEW

File: src/app/_utils/structuredData.ts [hunk 1]
```diff
@@ -17,13 +27,32 @@ export const baseOrganizationSchema: WithContext<Organization> = {
   '@type': 'Organization',
   name: ORGANIZATION_NAME,
   url: BASE_URL,
+  sameAs: [
+    'https://bsky.app/profile/ff.bsky.social',
+    'https://github.com/FilecoinFoundationWeb',
+    'https://www.linkedin.com/company/filecoin-foundation/mycompany/',
+    'https://twitter.com/FilFoundation',
+    'https://www.youtube.com/channel/UCeM5ezNgFM1Sle_gIz-KMMA',
+  ],
```
Comment (mirhamasala) [inline src/app/_utils/structuredData.ts:36]: We can use whatever we have under `FILECOIN_FOUNDATION_URLS`

File: src/app/_utils/structuredData.ts [hunk 2]
```diff
@@ -17,13 +27,32 @@ export const baseOrganizationSchema: WithContext<Organization> = {
...
+      '@type': 'ContactPoint',
+      contactType: 'Media and collaboration inquiries',
+      email: 'mailto:hello@fil.org',
+    },
+    {
+      '@type': 'ContactPoint',
+      contactType: 'Ecosystem grants inquiries',
+      email: 'mailto:devgrants@fil.org',
+    },
+  ],
```
Comment (mirhamasala) [inline src/app/_utils/structuredData.ts:55]: Same - Let's use `FILECOIN_FOUNDATION_URLS`

Comment (mirhamasala) [reply src/app/_utils/structuredData.ts:55]: We can make `email` an object with a label and whatever we have above under `contactType` - Please refactor the rest as well (No need to use label when it's different.)

File: src/app/_components/SiteLayout.tsx [hunk 3]
```diff
@@ -14,7 +11,6 @@ export function SiteLayout({ children }: LayoutProps) {
       <body className="m-auto flex max-w-[1032px] flex-col justify-between bg-brand-800 px-6 pb-6 pt-8 text-brand-100">
         <NetlifyIdentityManager />
```
Comment (mirhamasala) [inline src/app/_components/SiteLayout.tsx:13]: This was supposed to be a bit of a fallback - Do we have to remove this as to not have duplicates? Did we check if there's a way to keep this as a fallback without having duplicates?

File: src/app/_utils/structuredData.ts [hunk 4]
```diff
@@ -36,4 +65,228 @@ export const generateWebPageStructuredData = ({
...
+        name: title,
+        item: `${BASE_URL}${path}`,
+      },
+    ],
+  },
 })
+
+export const generateBlogPageStructuredData = (
+  posts: BlogPostData[],
+  seo: any,
```
Comment (mirhamasala) [inline src/app/_utils/structuredData.ts:89]: `any` is never a good option - I'm surprised that TS isn't yelling at you?

**Review by mirhamasala — CHANGES_REQUESTED** [review]
This is looking very good! Thank you so much for including screenshots of the test results. Super helpful. ✨ Here a few requests for improvement:

1. Let's use function declarations for improved readability and consistency with the rest of the codebase
2. I think it would be cool if we created local `utils` folders for each page - The same way we did for `data` per your suggestion. This way, we can keep the generating functions closer to the pages that need it. You can keep the shared utils where they are.

File: src/app/_utils/structuredData.ts (hunk 2 above)
Comment (barbaraperic) [reply src/app/_utils/structuredData.ts:55]: do you mean something like this? 

```
 grants: {
    github: 'https://github.com/filecoin-project/devgrants',
    email: {
      label: 'Ecosystem grants inquiries',
      address: 'mailto:devgrants@fil.org',
    },
  },
```

Comment (mirhamasala) [reply src/app/_utils/structuredData.ts:55]: Exactly! The only thing I would change is to rename `address` to `href`

File: src/app/_components/SiteLayout.tsx (hunk 3 above)
Comment (barbaraperic) [reply src/app/_components/SiteLayout.tsx:13]: Hmm I didn't find a way not have duplicates in the pages if we include that one. And I don't see anything [mentioned in the documentation.](https://nextjs.org/docs/app/building-your-application/optimizing/metadata#json-ld). We could included the checks in cypress test?

**Comment by barbaraperic** [comment]
> This is looking very good! Thank you so much for including screenshots of the test results. Super helpful. ✨ Here a few requests for improvement:
> 
> 1. Let's use function declarations for improved readability and consistency with the rest of the codebase
> 2. I think it would be cool if we created local `utils` folders for each page - The same way we did for `data` per your suggestion. This way, we can keep the generating functions closer to the pages that need it. You can keep the shared utils where they are.
> 3. The BreadCrumb name for Ecosystem page seems incorrect (at least reading from the screenshot.)

1. ✅
2. ✅
3. Do you mean it should be `ecosystem?sort=newest`?

File: src/app/_components/SiteLayout.tsx (hunk 3 above)
Comment (mirhamasala) [reply src/app/_components/SiteLayout.tsx:13]: Yes - We can include checks. Let's add a ticket to the backlog. 🙏🏼

**Comment by mirhamasala** [comment]
@barbaraperic - I don't think you understood this one

> Let's use function declarations for improved readability and consistency with the rest of the codebase

I've made the changes.

For 3. The right data wasn't being passed.

**Comment by barbaraperic** [comment]
@mirhamasala all good with the test 👌

EcosystemPost
PrivacyPolicy
EventPost
BlogPost
Governance
Ecosystem
Grants
Events
Terms
About
Blog
Home

**Comment by barbaraperic** [comment]
> @barbaraperic Thanks so much for checking! Do you happen to have seen what the non-critical issues are?

For `event/slug`
Screenshot 2024-06-05 at 20 56 32


For `blog/slug`
Screenshot 2024-06-05 at 20 58 36

**Review by mirhamasala — APPROVED** [review]

**Comment by mirhamasala** [comment]
@barbaraperic Thank you for your patience with this one. Let's get it merged. ❤️

**Pull request merged**